ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicaThread.java

```diff
@@ -112,11 +116,24 @@ public Map<DataNodeId, List<RemoteReplicaInfo>> selectReplicas(Map<DataNodeId, L
     replicas.values().forEach(rlist -> rlist.forEach(replica -> partitions.computeIfAbsent(
         replica.getReplicaId().getPartitionId().getId(), k -> new ArrayList<>()).add(replica)));
     // Pick one replica per partition for this iteration
-    // Group by data node
     Map<DataNodeId, List<RemoteReplicaInfo>> nodes = new HashMap<>();
     partitions.values().forEach(rlist -> {
       rlist.sort(comparator);
-      RemoteReplicaInfo replica = rlist.get(numReplIter % rlist.size());
+      RemoteReplicaInfo replica;
+      switch (vcrNodeConfig.replicaSelectionPolicy) {
+        case FIXED:
+          replica = rlist.get(0);
+          logger.trace("FIXED replicaSelectionPolicy picked {} for partition-{}",
+              replica.getReplicaId().getDataNodeId().getHostname(), replica.getReplicaId().getPartitionId().getId());
+          break;
+        case ROUND_ROBIN:
+        default:
+          replica = rlist.get(numReplIter % rlist.size());
+          logger.trace("{} replicaSelectionPolicy picked {} for partition-{}",
+              vcrNodeConfig.DEFAULT_REPLICA_SELECTION_POLICY, replica.getReplicaId().getDataNodeId().getHostname(),
+              replica.getReplicaId().getPartitionId().getId());
+      }
+      // Group by data node
```

Comment:
i just went back and checked out how different replicas of the same partition get added to replication thread. It seems that we would not put replicas from the same partition to one replica thread, they will away end up with different replica thread.